packages/trader/src/Modules/Trading/Containers/trade.jsx

```diff
@@ -31,10 +31,23 @@ class Trade extends React.Component {
         digits: [],
         tick: {},
         try_synthetic_indices: false,
+        try_open_markets: false,
+        category: null,
+        subcategory: null,
     };
 
-    componentDidMount() {
+    async componentDidMount() {
         this.props.onMount();
+        if (this.props.is_eu) {
+            const markets_to_search = ['forex', 'indices', 'commodities']; // none-synthetic
+            const { category, subcategory } = await this.props.getFirstOpenMarket(markets_to_search);
```

Comment:
There are cases where `getFirstOpenMarket` would return undefined. Wouldn't it result to uncaught TypeError?

Reply:
Yep. Fixed.